### ISSUE TITLE
Sorting 2

### ISSUE THREAD

**Issue by edd-cgfan** [softsyst/qirx_issues#104]
Hi Clem,

By chance, under special DX conditions, I wanted to know how to sort each transmitter site by distance:

![10B-rbb-TII-km-Sorting-problem-ascending](https://github.com/softsyst/qirx_issues/assets/49753409/b1095402-556c-47d3-84fc-05c158e15569)

![10B-rbb-TII-km-Sorting-problem-descending](https://github.com/softsyst/qirx_issues/assets/49753409/e3f337a6-2b54-490e-b269-d7992f23baa4)

What one would expect is this for the ascending case:
![10B-rbb-TII-km-Sorting-ok](https://github.com/softsyst/qirx_issues/assets/49753409/abdec8e3-1752-4052-a780-efb24470d798)

This is another sorting problem:
It is the km column of the TII tab.
It is not an issue of QIRX, it is an issue of the control (WPF DataGrid?) because the default sorting algorithm of the control doesn't sort numerically. 

Here is a link to a raw file containing this case:
[10B-rbb-raw-file](https://drive.google.com/file/d/1oWw7D_MUhUwOhGnvFMjHOiFpeFnj5Lh6/view?usp=sharing)

You can check it if you change the OwnPosition line in the qirx4.config file to:
`<OwnPosition Latitude="52.131489" Longitude="11.633654" />
`

Maybe you can think about a custom sorting for the km columns when you work on the other sorting problem (#102).

Regards,
cgfan

**Comment by softsyst** [comment]
Hi cgfan,

Your problem is well-known. The column sorting is a nasty issue, however in another way as the one you described. What you need is almost no effort. It has been "sloppy" programming by whoever created that code...
The WPF data grid handles it automatically in the correct way if you give it the correct data type. I gave it a string, so it sorted as a string... 
I made a quick test and gave it a numerical type (double) as datatype, and this is the result from your file (thanks for it btw), with the "km_abs" column showing the correct sorting:

![image](https://github.com/softsyst/qirx_issues/assets/51386375/6a7fbf5d-9063-44d5-83ca-5ca7ab939eae)

Then, where's the "nasty" problem? 
The problem is that you do not know which column is sorted, and in which way (ascending, descending). The "correct" and expected solution is to show some indication on the sorted column.
The xaml provides this also, but (and this is the nasty part) only in the following way:

![image](https://github.com/softsyst/qirx_issues/assets/51386375/3f1fe270-6449-483d-9adb-720135506878)

This really looks not nice at all.
As soon as you change the default background color, i.e. the one shown here, the sorting indication is gone. It comes back only if you write a template with many lines of code.
Whoever decided this at MS, ....

In any case, the correct sorting will be included in the next version. The sorting indication would be a separate issue.

BTW, this issue might affect also other tables, which I will investigate.

Regards,
Clem

**Comment by softsyst** [comment]
Should work in 4.1.1

**Comment by edd-cgfan** [comment]
Hi Clem,

Thanks for the new version and your efforts solving this sorting problem.
It works well.
I like the way how you show which column is sorted in which direction.

However, I found one thing: After the first start of the receiver and not having touched any column header:
Please see the images:
![Initial-sorting-5C](https://github.com/softsyst/qirx_issues/assets/49753409/2ea602ef-1ab1-469f-b8df-fb93396b7179)

At the following one it seems to be sorted right, but the the arrow for indication of the sorting isn't the right one:
![Initial-sorting-5C-3](https://github.com/softsyst/qirx_issues/assets/49753409/2e17faa3-31ba-4f6f-9b53-f3454339b315)
(After looking at the image again, I see that the sorting does not seem to be correct: 0.024 and 0.025)

Nevertheless, my suggestion:
Just omit the arrow and show it only after one has clicked on the column if that is easy to implement.

Regards,
cgfan

**Comment by softsyst** [comment]
Hi cgfan,

many thanks for this quick test. The behavior was (half) intended, as that table was not considered to need a permanent update of the sorting. However, in a dynamic environment this would be at least confusing. 

Therefore, I put an updated new version 4.1.2 into the "Download" which should fix this. I also considered your comment about showing the sorting triangle only after first user action.

The "History" page got a remark about the new version, nothing else.

Clem

**Comment by edd-cgfan** [comment]
Hello Clem,

Thanks for 4.1.2.
Table sorting looks very good now:
Works perfectly.

Regards,
cgfan

**Comment by softsyst** [comment]
Thanks for feedback. I close this.

Clem
